src/frontend/komponenter/Behandling/Brev/Brevmeny.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { Dispatch, useEffect, useState } from 'react';
+import {
+    BrevStruktur,
+    Flettefelter,
+    FlettefeltMedVerdi,
+    Flettefeltreferanse,
+    ValgFelt,
+    Valgmulighet,
+    ValgtFelt,
+} from './BrevTyper';
+import { BrevMenyDelmal } from './BrevMenyDelmal';
+import { byggTomRessurs, Ressurs } from '../../../typer/ressurs';
+import { RessursStatus } from '@navikt/familie-typer';
+import { finnFlettefeltNavnFraRef } from './BrevUtils';
+import { useApp } from '../../../context/AppContext';
+import DataViewer from '../../Felleskomponenter/DataViewer/DataViewer';
+import { Knapp } from 'nav-frontend-knapper';
+import styled from 'styled-components';
+import { IPersonopplysninger } from '../../../typer/personopplysninger';
+import { dagensDatoFormatert } from '../../../utils/formatter';
+
+interface Props {
+    settBrevRessurs: Dispatch<Ressurs<string>>;
+    behandlingId: string;
+    personopplysninger: IPersonopplysninger;
+}
+
+const GenererBrev = styled(Knapp)`
+    display: block;
+    margin: 0 auto;
+`;
+
+const StyledBrevmeny = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 12px;
+`;
+
+const Brevmeny: React.FC<Props> = ({ settBrevRessurs, behandlingId, personopplysninger }) => {
+    const { axiosRequest } = useApp();
+    const [brevStruktur, settBrevStruktur] = useState<Ressurs<BrevStruktur>>(byggTomRessurs());
+    const [alleFlettefelter, settAlleFlettefelter] = useState<FlettefeltMedVerdi[]>(
+        [] as FlettefeltMedVerdi[]
+    );
+    const [valgteFelt, settValgteFelt] = useState<{ [valgFeltKategori: string]: Valgmulighet }>(
+        {} as ValgtFelt
+    );
+    const [valgteDelmaler, settValgteDelmaler] = useState<{ [delmalNavn: string]: boolean }>({});
+
+    const brevMal = 'innvilgetOvergangsstonadHoved2';
+    const datasett = 'ef-brev';
+
+    useEffect(() => {
+        axiosRequest<BrevStruktur, null>({
+            method: 'GET',
+            url: `/familie-brev/api/${datasett}/avansert-dokument/bokmaal/${brevMal}/felter`,
+        }).then((respons: Ressurs<BrevStruktur>) => {
+            if (respons.status === RessursStatus.SUKSESS) {
```

Comment:
Burde man kunne dele opp denne komponenten i 2, en som ansvarer for å hente felter og en som håndterer resten av filen? Og bruke DataViewer med resultatet.
Hvis man får feil nå mot `/familie-brev/api/${datasett}/avansert-dokument/bokmaal/${brevMal}/felter` blir det aldrig vist.
Man kan då også slette `brevStruktur.status === RessursStatus.SUKSESS` på noen steder.

Reply:
Ja, det virker lurt 👌

Reply:
Enig i denne